### PR TITLE
Validate more static pods on masters

### PR DESCRIPTION
**Pull request by johngmyers**
This should cut down on cluster validation flapping.

Not testing the etcd pods as their `k8s-app` annotation depends on whether or not the cluster is configured to use etcd-manager. (It's also unlikely kube-apiserver would be ready without them.) Not testing kube-proxy pods because whether or not they're supposed to be there depends on the cluster spec.

Suppressing the static pod checks on non-ready nodes to cut down on redundant failure messages.
/area rolling-update

### CODE REVIEW

File: pkg/validation/validate_cluster.go [hunk 1]
```diff
@@ -199,13 +199,22 @@ func (v *ValidationCluster) collectComponentFailures(ctx context.Context, client
 	return nil
 }
 
+var masterStaticPods = []string{
+	"kube-apiserver",
+	"kube-controller-manager",
+	"kube-scheduler",
```
Comment (hakman) [inline pkg/validation/validate_cluster.go:205]: Just curious, do `etcd-manager-*` startic pods get checked also?

Comment (johngmyers) [reply pkg/validation/validate_cluster.go:205]: No. If the cluster uses the legacy etcd provider, there won't be any `etcd-manager-*` pods. Checking the cluster's spec would add a lot of complexity. I suspect the kube-apiserver pods won't be able to go into ready state without etcd pods.

Comment (hakman) [reply pkg/validation/validate_cluster.go:205]: OK, probably not worth the trouble for now. Thanks.

**Comment by justinsb** [comment]
Thanks @johngmyers  - looks great.  Not sure why we're not using spec.nodeName - maybe that is only populated in recent k8s versions?  (Do you remember?)

/approve
/lgtm

**Pull request merged**